main/pipeline/pipeline.cc

```diff
@@ -983,23 +987,30 @@ ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<
                                         igs.tracer().error("Exception typing file: {} (backtrace is above)",
                                                            file.data(igs).path());
                                     }
+                                    // Stream out errors
+                                    resultq->push(move(threadResult), processedByThread);
+                                    processedByThread = 0;
                                 }
                             }
                         }
                     }
                     if (processedByThread > 0) {
-                        threadResult.counters = getAndClearThreadCounters();
                         resultq->push(move(threadResult), processedByThread);
                     }
                 });
 
+            if (workers.size() > 0) {
```

Comment:
This needs to go after the loop below, and you need a second loop that pops counters off of `counterQueue` and calls `counterConsume` on them.